test/audio-playerSpec.js

```diff
@@ -0,0 +1,279 @@
+const AudioPlayer = require('../audio-player');
+const CordovaMediaPlayer = require('../cordova-media-player');
+const WebAudioPlayer = require('../web-audio-player');
+
+describe('Audio player', () => {
+  const oldMedia = window.Media;
+  const oldAudio = window.Audio;
+
+  afterAll(() => {
+    window.Media = oldMedia;
+    window.Audio = oldAudio;
+  });
+
+  beforeEach(() => {
+    spyOn(AudioPlayer.prototype, '_playbackCompatibility');
+    spyOn(AudioPlayer.prototype, '_getBestPlayer');
+  });
+
+  it('should construct with event functionality', () => {
+    const player = new AudioPlayer();
+    player.emitter = {
+      on: jasmine.createSpy(),
+      off: jasmine.createSpy(),
+      emit: jasmine.createSpy()
+    };
+    window.allOff = jasmine.createSpy();
```

Comment:
allOff is function imported from the event-emitter package in audio-player.js

`const allOff = require('event-emitter/all-off');`